Added systemd service file and refactored main.py

KJSain opened the pull request:
Hello!

First time open-source contributing. Let me know if I'm doing this wrong.

I switched the main.py to use the argparse library to facilitate easier argument parsing. I also added the service file that I created to use for Linux instead of using screen to offload the backgrounding of the service. I think this makes it more "Linux-y".

- Switched main.py to use argparse for cleaner code and easier argument additions for the future 
- Added extras folder with systemd service file example

jaqb committed:
Added support for local trakt list

KJSain committed:
Added systemd service file and refactored main.py

Switched main.py to use argparse for cleaner code and easier argument additions for the future
Added extras folder with systemd service file example

dreulavelle reviewed (approved):
This is a neat feature! Awesome work.

sambanks commented:
Thanks heaps for this!

I didn't like the screen bits of the install guide so started doing this, then thought to check the issues.

Works great!

sirstudly committed:
Resolving issue https://github.com/itsToggle/plex_debrid/issues/638

sirstudly committed:
Allow movie matches with year NOT present in title
Also allow .avi files
eg. The.Wild.Parrots.Of.Telegraph.Hill.XviD.AC3

[29/03/24 01:46:08] scraping sources [torrentio] for query "the.wild.parrots.of.telegraph.hill.2005" ... done
[29/03/24 01:46:08] accepting titles that regex match "[^A-Za-z0-9]*((the.wild.parrots.of.telegraph.hill):?.*)\(?\[?(2005)(tt0424565)?" ... done - found 3 releases
[29/03/24 01:46:08] scraping sources [torrentio] for IMDB ID "tt0424565" ... done
[29/03/24 01:46:08] accepting titles that regex match "(.*|tt0424565)" ... done - found 3 releases
[29/03/24 01:46:13] checking cache status for scraped releases on: [Real Debrid] ... done
[29/03/24 01:46:14] [realdebrid] checking and sorting all release files ... done
[29/03/24 01:46:14] set release bitrate using total movie duration: 01h:23m
[29/03/24 01:46:14] 1) title: The.Wild.Parrots.Of.Telegraph.Hill.XviD.AC3           | size: 1.02 | bitrate: 2.05 | cached: RD | seeders: 9 | files: +0/-0 | source: [torrentio: MagnetDL]
[29/03/24 01:46:14] 2) title: Wild.Parrots.of.Telegraph.Hill.(DVD.Rip).rar          | size: 0.87 | bitrate: 1.75 | cached:    | seeders: 0 | files: +0/-0 | source: [torrentio: ThePirateBay]
[29/03/24 01:46:14] 3) title: Debbie.Does.Dallas.Uncovered.2005.DVDRip.X264.Konnann | size: 0.79 | bitrate: 1.59 | cached: RD | seeders: 2 | files: +2/-0 | source: [torrentio: MagnetDL]
[29/03/24 01:46:14] 4) title: The.Wild.Parrots.Of.Telegraph.Hill.XviD.AC3           | size: 1.02 | bitrate: 2.05 | cached: RD | seeders: 9 | files: +0/-0 | source: [torrentio: MagnetDL]
[29/03/24 01:46:14] 5) title: Wild.Parrots.of.Telegraph.Hill.(DVD.Rip).rar          | size: 0.87 | bitrate: 1.75 | cached:    | seeders: 0 | files: +0/-0 | source: [torrentio: ThePirateBay]
[29/03/24 01:46:14] 6) title: Debbie.Does.Dallas.Uncovered.2005.DVDRip.X264.Konnann | size: 0.79 | bitrate: 1.59 | cached: RD | seeders: 2 | files: +2/-0 | source: [torrentio: MagnetDL]
[29/03/24 01:46:14] sorting releases for version [1080p SDR] ... done - found 2 releases
[29/03/24 01:46:14] 1) title: The.Wild.Parrots.Of.Telegraph.Hill.XviD.AC3 | size: 1.02 | bitrate: 2.05 | cached: RD | seeders: 9 | files: +0/-0 | source: [torrentio: MagnetDL]
[29/03/24 01:46:14] 2) title: The.Wild.Parrots.Of.Telegraph.Hill.XviD.AC3 | size: 1.02 | bitrate: 2.05 | cached: RD | seeders: 9 | files: +0/-0 | source: [torrentio: MagnetDL]
[29/03/24 01:46:14] [realdebrid] error: rejecting release: "The.Wild.Parrots.Of.Telegraph.Hill.XviD.AC3" because it doesnt match the allowed deviation "[^A-Za-z0-9]*((the.wild.parrots.of.telegraph.hill):?.*)\(?\[?(2005|2004|2006)"
[29/03/24 01:46:14] [realdebrid] error: rejecting release: "The.Wild.Parrots.Of.Telegraph.Hill.XviD.AC3" because it doesnt match the allowed deviation "[^A-Za-z0-9]*((the.wild.parrots.of.telegraph.hill):?.*)\(?\[?(2005|2004|2006)"
[29/03/24 01:46:14] retrying download in 30min for item: the.wild.parrots.of.telegraph.hill.2005 - version/s [1080p SDR] - attempt 1/5

sirstudly committed:
Fix issue 578: Reoccurrence of Issue #478: Single Episode Downloads Not Functioning Correctly

sirstudly committed:
Fix issue 578: Reoccurrence of Issue #478: Single Episode Downloads Not Functioning Correctly

sirstudly committed:
classes.py:deviation(): improve regex matching
- handle ampersands in titles, eg. Fast.Cheap.&.Out.of.Control.a.film.by.Errol.Morris
- incorrect matching at beginning of title, eg.  [realdebrid] error: rejecting release: "Knjiga.rekorda.Sutke.AKA.The.Shutka.Book.Of.Records" because it doesnt match the allowed deviation "[^A-Za-z0-9]*((the.shutka.book.of.records):?.*)\(?\[?(2005|2004|2006)?"

sirstudly committed:
Parameterized watchlist loop interval rather than hardcoded 30 minutes

sirstudly committed:
Parameterized watchlist loop interval rather than hardcoded 30 minutes

sirstudly committed:
Fix deviation() for season release of:
Land.of.the.Giants.1968.Season.1[Complete]

petern-sc committed:
Add plex server address help

- Plex doesn't like extra trailing slashes and requests requires the protocol for urls

funkypenguin committed:
"Elfify" the logo

Signed-off-by: David Young <davidy@funkypenguin.co.nz>

funkypenguin committed:
try to pass oauth details to PD using env vars

Signed-off-by: David Young <davidy@funkypenguin.co.nz>

sirstudly committed:
Added Zilean support

sirstudly committed:
Added Zilean support

sirstudly committed:
Updated docker files

sirstudly committed:
Additional comments and debugging statements

sirstudly committed:
Fix missing originallyAvailableAt on classes::released

sirstudly committed:
Issue 529: Update default character replacement to better handle punctuation

sirstudly committed:
Fixing zilean support

sirstudly committed:
Allow access to host network in docker-compose.yml

sirstudly committed:
Fixing zilean alt matching

sirstudly committed:
Fixing zilean alt matching

sirstudly committed:
Fixing zilean alt matching

funkypenguin committed:
Merge pull request #1 from sirstudly/issue_578

Fix issue 578/478: Single Episode Downloads Not Functioning Correctly

funkypenguin committed:
Merge pull request #3 from sirstudly/zilean_support

Added Zilean support

funkypenguin committed:
Merge pull request #2 from sirstudly/watchlist_loop_param

Parameterized watchlist loop interval rather than hardcoded 30 minutes

funkypenguin committed:
Bump version to v2.96

Signed-off-by: David Young <davidy@funkypenguin.co.nz>

funkypenguin committed:
Merge branch 'main' of github.com:elfhosted/plex_debrid

sirstudly committed:
Issue 529: Update default character replacement to better handle punctuation

Correct matching for:
  - Mission Impossible: Dead Reckoning
    - "Part One" may be present in name but not matched
  - Mr. Nobody
    - Allow for Mr..Nobody or Mr.Nobody
  - Dirk Gently's Holistic Detective Agency
    - Allow presence of apostrophes in title
  - Face/Off
    - Allow for slashes in title

Squashed commit of the following:

commit de12f022c75b0d194441f4f7690b213016b3edc8
Author: sirstudly <ron1404@techie.com>
Date:   Tue Jul 23 17:08:51 2024 +0100

    Character replacement for slashes in title (eg. Face/Off)
    Issue #529

commit 5fbb9991895180c92b27d6114062aa342b827dd5
Author: sirstudly <ron1404@techie.com>
Date:   Wed Jul 17 22:54:14 2024 +0100

    Issue 529: Update default character replacement to better handle punctuation

commit 113caebffea6961dcf4fef4536ee494628ab20b7
Author: sirstudly <ron1404@techie.com>
Date:   Thu Apr 4 18:42:42 2024 +0100

    Keeping regex consistent with subsequent line (issue 529)

commit fedbc673366a5b4f7ba9bd45ce4fbbfbefbe6f98
Author: sirstudly <ron1404@techie.com>
Date:   Thu Mar 21 13:49:34 2024 +0000

    Resolving issue https://github.com/itsToggle/plex_debrid/issues/638

sirstudly committed:
Merge branch 'issue_529'

sirstudly committed:
Reverting some regex fixes which may not be required?

sirstudly committed:
Reverting previous commit.. included in branch issue_529

KJSain committed:
Merge pull request #1 from sirstudly/main

pull sirstudly's pull request into forked repo

KJSain committed:
Merge branch 'main' into pr/684

KJSain committed:
Merge pull request #3 from jaqb/main

Added support for local trakt list
